GoogleTestAdapter/Core/Runners/CommandLineGenerator.cs

```diff
@@ -69,39 +69,59 @@ private IEnumerable<Args> GetFinalCommandLines(string baseCommandLine)
                 return commandLines;
             }
 
-            string baseCommandLineWithFilter = baseCommandLine + GoogleTestConstants.FilterOption;
-
             List<string> suitesRunningAllTests = GetSuitesRunningAllTests();
-            baseCommandLineWithFilter += GetFilterForSuitesRunningAllTests(suitesRunningAllTests);
-
-            List<TestCase> testCasesNotRunBySuite = GetTestCasesNotRunBySuite(suitesRunningAllTests);
-            List<TestCase> testCasesRunBySuite = _testCasesToRun.Where(tc => !testCasesNotRunBySuite.Contains(tc)).ToList();
-            if (testCasesNotRunBySuite.Count == 0)
+            int suiteLengthAggregate = 0;
+            int groupIdxCounter = 0;
+            int maxSuiteLength = MaxCommandLength - _lengthOfExecutableString - userParam.Length - 1;
+
+            var suiteGroups = suitesRunningAllTests
```

Comment:
This is some neat usage of Linq :-) However, I feel that we are kind of mixing abstraction levels here. I would recommend moving this code (i.e., lines 74 to 87?) into an appropriately named method.